src/particles/deposition/ExplicitDeposition.cpp

```diff
@@ -192,38 +174,37 @@ ExplicitDeposition (PlasmaParticleContainer& plasma, Fields& fields, const Multi
                             AabssqDyp = (x00yp1-x00ym1) * 0.5_rt * dy_inv * laser_fac * clight;
                         }
 
-                        amrex::Gpu::Atomic::Add(arr.ptr(i, j, Sy),
+                        amrex::Gpu::Atomic::Add(arr.ptr(i, j, Sy), charge_density_mu0 * (
                             - shape_x * shape_y * (
                                 - Bz_v * vx
                                 + ( Ez_v * vy
                                 + ExmBy_v * (          - vx * vy)
-                                + EypBx_v * (gamma_psi - vy * vy) ) / clight
-                                - 0.25_rt * AabssqDyp * q_mass_ratio / psi
-                            ) * q_mass_ratio / psi
-                            - shape_dx * shape_y * (
+                                + EypBx_v * (gamma_psi - vy * vy) ) * clight_inv
+                                - 0.25_rt * AabssqDyp * q_mass_ratio * psi_inv
+                            ) * q_mass_ratio * psi_inv
+                            + ( - shape_dx * shape_y * dx_inv * (
                                 - vx * vy
                             )
-                            - shape_x * shape_dy * (
+                            - shape_x * shape_dy * dy_inv * (
                                 gamma_psi - vy * vy - 1._rt
-                            )
-                        );
+                            )) * clight
+                        ));
 
-                        amrex::Gpu::Atomic::Add(arr.ptr(i, j, Sx),
+                        amrex::Gpu::Atomic::Add(arr.ptr(i, j, Sx), charge_density_mu0 * (
                             + shape_x * shape_y * (
                                 + Bz_v * vy
                                 + ( Ez_v * vx
                                 + ExmBy_v * (gamma_psi - vx * vx)
-                                + EypBx_v * (          - vx * vy) ) / clight
-                                - 0.25_rt * AabssqDxp * q_mass_ratio / psi
-                            ) * q_mass_ratio / psi
-                            + shape_dx * shape_y * (
+                                + EypBx_v * (          - vx * vy) ) * clight_inv
+                                - 0.25_rt * AabssqDxp * q_mass_ratio * psi_inv
+                            ) * q_mass_ratio * psi_inv
+                            + ( + shape_dx * shape_y * dx_inv * (
                                 gamma_psi - vx * vx - 1._rt
                             )
-                            + shape_x * shape_dy * (
+                            + shape_x * shape_dy * dy_inv * (
                                 - vx * vy
-                            )
-                        );
-
+                            )) * clight
+                        ));
```

Comment:
A few factors changed here (`charge_density_mu0`, `dy_inv`, etc), could you explain quickly why that is?

Reply:
Reducing the number of floating-point divisions helps improve performance greatly. Some other factors where reshuffled to (slightly) reduce register usage.